encodings/fsst/tests/fsst_tests.rs

```diff
@@ -0,0 +1,149 @@
+#![cfg(test)]
+
+use arrow_array::builder::BinaryBuilder;
+use fsst::{Compressor, Symbol};
+use vortex::array::{BoolArray, PrimitiveArray};
+use vortex::arrow::FromArrowArray;
+use vortex::compute::unary::scalar_at;
+use vortex::compute::{filter, slice, take};
+use vortex::validity::Validity;
+use vortex::{ArrayDType, ArrayDef, IntoArray};
+use vortex_dtype::{DType, Nullability, PType};
+use vortex_fsst::{FSSTArray, FSST};
+
+fn build_fsst_utf8_array() -> FSSTArray {
+    let compressor = Compressor::train(
+        "The Greeks never said that the limit could not he overstepped. \
+        They said it existed and that whoever dared to exceed it was mercilessly struck down. \
+        Nothing in present history can contradict them.",
+    );
+
+    let symbols = compressor.symbol_table();
+
+    // SAFETY: Symbol and u64 have same size, enforced by compiler
+    let symbols_u64 = unsafe { std::mem::transmute::<&[Symbol], &[u64]>(symbols) };
+    let mut symbols_vec = Vec::new();
+    symbols_vec.extend_from_slice(symbols_u64);
+
+    let symbols_array = PrimitiveArray::from_vec(symbols_vec, Validity::NonNullable).into_array();
+    assert_eq!(
+        symbols_array.dtype(),
+        &DType::Primitive(PType::U64, Nullability::NonNullable)
+    );
+
+    let mut codes = BinaryBuilder::new();
+
+    codes.append_value(
+        compressor
+            .compress("The Greeks never said that the limit could not he overstepped".as_bytes()),
+    );
+    codes.append_value(
+        compressor.compress(
+            "They said it existed and that whoever dared to exceed it was mercilessly struck down"
+                .as_bytes(),
+        ),
+    );
+    codes.append_value(
+        compressor.compress("Nothing in present history can contradict them".as_bytes()),
+    );
+
+    let codes = codes.finish();
+    let codes_array = vortex::Array::from_arrow(&codes, false);
+
+    FSSTArray::try_new(
+        DType::Utf8(Nullability::NonNullable),
+        symbols_array,
+        codes_array,
+    )
+    .expect("building from parts must succeed")
+}
+
+macro_rules! assert_nth_scalar {
+    ($arr:expr, $n:expr, $expected:expr) => {
+        assert_eq!(scalar_at(&$arr, $n).unwrap(), $expected.try_into().unwrap());
+    };
+}
+
+#[test]
+fn test_compute() {
```

Comment:
wdyt about splitting this test into dedicated tests? Seems like its just easier to hunt bugs/expaned/refactor things that way, and failures are easier to narrow down.
You can use [`rstest`](https://docs.rs/rstest/latest/rstest/) to make `fsst_array` into a fixture that can be shared between them.

Reply:
also seems like this tests takes a pretty long time with miri.

Reply:
i've been having issues getting miri to run on FSST, i'll flag these off for now, will re-enable miri on them once i've fixed the issue library-side